qark/scanner/scanner.py

```diff
@@ -60,6 +61,28 @@ def _run_manifest_checks(self):
 
             self.issues.extend(plugin.issues)
 
+    def _run_intent_checks(self):
```

Comment:
Starting to feel like we should write a generic run checks function, and pass in the category as an argument. Feels like the 4th time I've seen this code snippet in the last couple days

Reply:
I agree with you, I think for now each plugin category should get its own method for this right now and then we will collapse them at the end if we decide that none of them need their own method. We didn't want to do this from the start because we were unsure what each plugin was doing specifically and what it required.